kubernetes/istio.go

```diff
@@ -175,6 +177,21 @@ func (in *K8SClient) CanConnectToIstiod() (IstioComponentStatus, error) {
 	return ics, nil
 }
 
+func (in *K8SClient) HasIstioAccess() bool {
+	componentStatus, err := in.CanConnectToIstiod()
+	if err != nil {
+		return false
+	}
+	if len(componentStatus) == 0 {
+		return false
+	}
+	return true
+}
```

Comment:
I think this would be better as a config option for the registry where you can enable/disable the registry altogether if you'd like rather than having it just be a fallback option: https://github.com/kiali/kiali/blob/master/config/config.go#L208.

@hhovsepy may be looking into this but you should be able to get the istio config from the kube api rather than the registry when there's no istiod access. 

What will be the impact on the controlplane namespace card once there is no direct istiod access? It might still be possible to get some details, like controlplane revision, about the controlplane from the injection webhook since this may be present in the cluster even if the controlplane is not.

Reply:
Funny, I was just looking at the same thing.  I also think it makes sense to configure Kiali for this scenario.  By default Kiali would expect to be able to connect to istiod's debug endpoint and proceed normally.  If it can't connect then that would be a FAILURE scenario for the control plane card.  If configured to ignore istiod then it would be a DEGRADED scenario for the control plane card (even if it is accessible we won't connect, this could be for performance reasons, or whatever).

Reply:
draft solution of retrieving configs when istiod is not available https://github.com/kiali/kiali/pull/5649

Reply:
> draft solution of retrieving configs when istiod is not available #5649

Thanks, @hhovsepy !